promdump/promdump.go

```diff
@@ -368,7 +376,118 @@ func exportMetric(ctx context.Context, promApi v1.API, metric string, beginTS ti
 			}
 		}
 	}
-	return writeFile(&values, filePrefix, fileNum)
+
+	return fileNum, writeFile(&values, filePrefix, fileNum)
+
+}
+
+// Holds custom metric file counts
+var customMetricCount uint
+
+func createArchive(buf io.Writer) error {
+	// Create new Writers for gzip , bzip2 and none(tar)
+
+	var tw *tar.Writer
+
+	switch *tarCompression {
+	case "gzip":
+		gw := gzip.NewWriter(buf)
+		defer gw.Close()
+		tw = tar.NewWriter(gw)
+
+	case "bzip2":
+		bw, err := bzip2.NewWriter(buf, nil)
+		if err != nil {
+			return err
+		}
+		defer bw.Close()
+		tw = tar.NewWriter(bw)
+
+	case "none":
+		tw = tar.NewWriter(buf)
+	default:
+		fmt.Errorf("unsupported compression type: %s , using default tar_compression_algorithm", *tarCompression)
```

Comment:
Should there be a `return` here? I also recommend using %v rather than %s most of the time.

Reply:
yes I miss to add return, edited  and changed %s to %v